pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -100,6 +102,17 @@ func NewCollector(cfg *Config, statesInformer statesinformer.StatesInformer, met
 	if c.config == nil {
 		c.config = NewDefaultConfig()
 	}
+	if ready := initNvml(); ready {
+		// init gpu device.
+		c.context.gpuDeviceManager = &gpuDeviceManager{}
```

Comment:
please mv `c.context.gpuDeviceManager = &gpuDeviceManager{}` to newCollectContext